examples/mongodb.yaml

```diff
@@ -0,0 +1,91 @@
+apiVersion: templating.flanksource.com/v1
+kind: Template
+metadata:
+  name: mongo-db
+spec:
+  source:
+    apiVersion: db.flanksource.com/v1
+    kind: MongoDB
+  resources:
+    - apiVersion: v1
+      kind: Secret
+      metadata:
+        name: mongodb-{{ .metadata.name }}-credentials
+        namespace: '{{ .metadata.namespace }}'
+      type: Opaque
+      stringData:
+        password: '{{- kget (print "secret/" .metadata.namespace "/mongodb-" .metadata.name "-credentials") "password" | default (random.AlphaNum 16) -}}'
+        admin_password: '{{- kget (print "secret/" .metadata.namespace "/mongodb-" .metadata.name "-credentials") "admin_password" | default (random.AlphaNum 16) -}}'
+
+    - apiVersion: apps/v1
+      kind: StatefulSet
+      metadata:
+        name: mongodb-{{ .metadata.name }}
+        namespace: '{{ .metadata.namespace }}'
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app.kubernetes.io/name: mongodb
+            app.kubernetes.io/instance: '{{ .metadata.name }}'
+        template:
+          metadata:
+            labels:
+              app.kubernetes.io/name: mongodb
+              app.kubernetes.io/instance: '{{ .metadata.name }}'
+          spec:
+            securityContext:
+              fsGroup: 184
+            containers:
+              - image: docker.io/centos/mongodb-36-centos7:1
+                name: mongodb
+                env:
+                  - name: MONGODB_DATABASE
+                    value: '{{ .spec.database }}'
+                  - name: MONGODB_USER
+                    value: '{{ .spec.user }}'
+                  - name: MONGODB_PASSWORD
+                    valueFrom:
+                      secretKeyRef:
+                        name: mongodb-{{ .metadata.name }}-credentials
+                        key: password
+                  - name: MONGODB_ADMIN_PASSWORD
+                    valueFrom:
+                      secretKeyRef:
+                        name: mongodb-{{ .metadata.name }}-credentials
+                        key: admin_password
+                ports:
+                  - containerPort: 27017
+                resources:
+                  requests:
+                    memory: 64Mi
+                    cpu: 50m
+                  limits:
+                    memory: 300Mi
+                    cpu: 100m
```

Comment:
these need to be templated

Reply:
What do you mean? This example would use the PodResource. Otherwise is the postgresdb example incorrect too?

Reply:
I think you looked at the postgres sidecar and not the main container.  It nneds to looklike:
https://github.com/flanksource/template-operator-library/blob/e4279cbe561875d83c2c90aafd9f5f23c9f79169/config/templates/postgresql-db.yaml#L85 (not like line 117)